### DISCUSSION TITLE
3.8.0

### DISCUSSION THREAD

**Discussion by Jolg42** [prisma/prisma#11099] [Announcements]
Today, we are excited to share the `3.8.0` stable release 🎉

🌟 **Help us spread the word about Prisma by starring the repo or [tweeting](https://twitter.com/intent/tweet?text=Check%20out%20the%20latest%20@prisma%20release%20v3.8.0%20%F0%9F%9A%80%0D%0A%0D%0Ahttps://github.com/prisma/prisma/releases/tag/3.8.0) about the release.** 🌟 

### Full-text search support for MySQL is now in Preview 🚀

We're back from the holidays with a special treat for you.

Prisma now supports full-text search in MySQL. You can enable full-text support by adding the `fullTextIndex` and `fullTextSearch` Preview flags in your Prisma schema and defining `@@fulltext()` indexes on fields you'd like to use full-text search on.

```prisma
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["fullTextIndex", "fullTextSearch"]
}

datasource db {
  provider = "mysql"
  url      = env("DATABASE_URL")
}

model Post {
  id     Int    @id @default(autoincrement())
  title  String @unique
  @@fulltext([title])
}
```

Run `prisma db push` or `prisma migrate dev` to update your database schema. Prisma Client will also be re-generated, enabling you to use full-text search in your application.

```typescript
// search for titles that contain cat, but not fox
await prisma.post.findMany({
  where: {
    title: {
      search: "+cat -fox",
    },
  },
})
```

Learn more in [our documentation](https://www.prisma.io/docs/concepts/components/prisma-client/full-text-search).

### `dataProxy` and `interactiveTransactions` Preview Features are now mutually exclusive

Before Prisma `3.8.0`, Prisma `$transaction` queries would fail whenever the Data Proxy and Interactive Transactions Preview features were used together. The `interactiveTransactions` and `dataProxy` Preview flags cannot be used together in this release. Generating the Prisma Client when both Preview features are enabled will throw an error.

### Fixed support for `push` when adding an element to an array in MongoDB

In `3.8.0`, we fixed [`push`](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#push) support for `ObjectId`s on MongoDB.

Given the following schema:

```prisma
model Course {
  id          String   @id @default(dbgenerated()) @map("_id") @db.ObjectId
  title       String
  students    String[] @db.Array(ObjectId)
}
```

You can now run the following query:

```typescript
// Add a new student to the course
await prisma.course.update({
  where: {
    id: 1
  },
  data: {
    students: {
      push: new ObjectID("...")
    }
  }
})
```

A special thanks to [Anthony Luzquiños](https://github.com/AnthonyLzq) for raising this issue!

### Prisma Migrate

- [@map field name swap causes "Field is already defined on model" error](https://github.com/prisma/prisma/issues/8687)
- [`generator.engineType` is not validated](https://github.com/prisma/prisma/issues/9120)
- [Migrate creates unnecessary migration with SQL syntax error](https://github.com/prisma/prisma/issues/9204)
- [Using datasource property that requires preview feature, highlights only value and not property in validation error](https://github.com/prisma/prisma/issues/9314)
- [Index `name` validations](https://github.com/prisma/prisma/issues/10082)
- [Postgres - 63 chars limit for index name](https://github.com/prisma/prisma/issues/10094)
- [IE: Not recognizing string type](https://github.com/prisma/prisma/issues/10347)
- [`previewFeatures = []` gets lost after Introspection](https://github.com/prisma/prisma/issues/10460)
- [Prisma format adds invalid fragments to schema if composites are present on a model](https://github.com/prisma/prisma/issues/10845)
- [`migrate dev` fails on second run with specific schema](https://github.com/prisma/prisma/issues/10983)
- [Error: [C:\\.cargo\registry\src\github.com-1ecc6299db9ec823\tiberius-0.6.4\src\tds\codec\pre_login.rs:129:22] unsupported prelogin token: 2 ](https://github.com/prisma/prisma/issues/11009)

### Prisma Client

- [MDBE: Design document for adding `type` support to the Query Engine](https://github.com/prisma/prisma/issues/9360)
- [Changing prisma query params using middleware doesn't work](https://github.com/prisma/prisma/issues/9522)
- [Implement Mysql Fulltext Querying Capabilities](https://github.com/prisma/prisma/issues/10415)
- [MongoDB referentialActions in m-n relationships](https://github.com/prisma/prisma/issues/10498)
- [`Error querying the database: db error: ERROR: unsupported startup parameter: options`](https://github.com/prisma/prisma/issues/10841)
- [How to push a new ObjectId to an array?](https://github.com/prisma/prisma/issues/10906)
- [`@prisma/client` intermittently added and removed from `transitivePeerDependencies` in lockfile](https://github.com/prisma/prisma/issues/10973)

### Language tools (e.g. VS Code)

- [Provider-aware auto-completion is not applied to all auto completions](https://github.com/prisma/language-tools/issues/942)
- [Add configuration option to turn off file watcher](https://github.com/prisma/language-tools/issues/944)
- [VSCode plugin formatting no longer uses editor `tabSize` configuration when indenting blocks](https://github.com/prisma/language-tools/issues/993)

### Prisma Studio

- [warn(prisma-client) Already 10 Prisma Clients are actively running.](https://github.com/prisma/studio/issues/800)
- [`BigInt` breaks studio](https://github.com/prisma/studio/issues/802)
- [Model named "@@id"](https://github.com/prisma/studio/issues/806)

## Credits

Huge thanks to @hyochan, @cesconix for helping!

## 📺 Join us for another "What's new in Prisma" livestream

Learn about the latest release and other news from the Prisma community by joining us for another ["What's new in Prisma"](https://www.youtube.com/watch?v=c236Qv1wBIU) livestream.

The stream takes place [on YouTube](https://www.youtube.com/watch?v=c236Qv1wBIU) on **Thursday, January 13** at **5 pm Berlin | 8 am San Francisco**.

<hr /><em>This discussion was created from the release <a href='https://github.com/prisma/prisma/releases/tag/3.8.0'>3.8.0</a>.</em>

**Comment by DanyylSocol** [comment]
Hi!
Release is really nice!
But i have problem - I need both dataProxy and interactiveTransactions
Is there analog to $transaction call?
Thank you!

**Reply by Jolg42** [reply]
Hi!
So if using `$transaction` without the Preview feature `interactiveTransactions` is not an option for you,
you can let the team a comment on https://github.com/prisma/prisma/issues/9841
